homeassistant/components/tasmota/__init__.py

```diff
@@ -40,6 +40,9 @@
 
 async def async_setup(hass: HomeAssistantType, config: dict):
     """Set up the Tasmota component."""
+    if "mqtt" not in hass.config.components:
```

Comment:
We need this check in `async_setup_entry` too since the component is only set up once, but the entry can be removed and set up again later.